bookchain-miner/src/block.cpp

```diff
@@ -0,0 +1,33 @@
+#include "block.hpp"
+
+Bloock::Bloock(std::string prevHash, std::string seedHash, int64_t height) :
```

Comment:
Any particular reason for omitting the hash of the block itself?

Reply:
If you add the hash of the block into the block the hash of the block will change, resulting in an infinite loop.

I think it comes down to weather the hash is of the block object itself or the data arranged in such a way, and how we want to deal with that.

Reply:
Yeah I see what you mean.  Thought we might do it like how we did it in the hackathon where the hash of a block is hash(prevBlockHash+data+nonce) so the hash of a particular block is independent of itself.  Not sure what the convention is here though.

Reply:
I think we should take advantage of the fact that in C++ we can store it already packed, having to set up the block structure each time you want to hash will lead to bugs and code duplication I feel.

I've updated it with a sub-block style thing that's kinda the best of both worlds.

Reply:
Ya that makes sense, sounds good 👍